pkg/fileutil/io.go

```diff
@@ -131,3 +136,34 @@ func FileExists(p string) (bool, error) {
 	}
 	return false, fmt.Errorf("%s: %w", p, ErrNotFile)
 }
+
+func VerifyAbsPath(absPath, basePath string) error {
+	// check we have a valid abs path
+	if !path.IsAbs(absPath) || path.Clean(absPath) != absPath {
+		return ErrBadPath
+	}
+	// point to storage namespace
+	if !strings.HasPrefix(absPath, basePath) {
+		return ErrInvalidPath
+	}
+	return nil
+}
+
+func VerifyRelPath(relPath, base string) error {
+	abs := base + string(os.PathSeparator) + relPath
+	return VerifyAbsPath(abs, base)
+}
+
+func VerifyRegularFile(name, base string) error {
+	filename, err := filepath.EvalSymlinks(name)
```

Comment:
I don't understand.  My understanding of the args is that if `name` is not an absolute path, it will be treated as `base + "/" + name`.  But in that case I'm confused about how this line should work.

Reply:
name can be either absolute or relative. According to the documentation of EvalSymlink:

> // EvalSymlinks returns the path name after the evaluation of any symbolic
// links.
// If path is relative the result will be relative to the current directory,
// unless one of the components is an absolute symbolic link.
// EvalSymlinks calls Clean on the result.

Reply:
True - removing basePath